src/strategies/MangroveOffer.sol

```diff
@@ -139,18 +139,11 @@ abstract contract MangroveOffer is AccessControlled, IOfferLogic {
   }
 
   /// @inheritdoc IOfferLogic
-  function setReserve(address maker, address reserve_) public override onlyCaller(maker) {
-    require(__checkReserveApproval__(reserve_, maker), "mgvOffer/makerNotApproved");
+  function setReserve(address maker, address reserve_) public override onlyAdmin {
```

Comment:
Previously admin could not fiddle with reserves - Is it safe to allow? Could admin misuse this? For instance set reserve to 0xcoffee and make all orders fail, collecting bounty?
Is reserve always either maker OR router, then maybe we could codify that somehow? For instance by inheriting from a building block that enforces desired behavior ?

Reply:
Yes this gives rugpull power to admin. I am wondering what is the best solution here. Maybe to set reserve at deploy time only?

Reply:
You fixed this by making the __reserve__ hook.